pmd-core/src/main/java/net/sourceforge/pmd/lang/LanguageRegistry.java

```diff
@@ -129,48 +116,4 @@ public static LanguageVersion findLanguageVersionByTerseName(String terseNameAnd
         return languages;
     }
 
-    public static List<LanguageVersion> findAllVersions() {
-        List<LanguageVersion> versions = new ArrayList<>();
-        for (Language language : getLanguages()) {
-            versions.addAll(language.getVersions());
-        }
-        return versions;
-    }
-
-    /**
-     * A utility method to find the Languages which have Rule support.
-     *
-     * @return A List of Languages with Rule support.
-     *
-     * @deprecated This method will be removed with PMD 7.0.0. Use {@link #getLanguages()} instead.
-     */
-    @Deprecated
-    public static List<Language> findWithRuleSupport() {
-        return new ArrayList<>(getLanguages());
-    }
-
-    public static String commaSeparatedTerseNamesForLanguage(List<Language> languages) {
```

Comment:
the one thing I don't like about loosing this method is that it's the `LanguageRegistry` the one place were we know we use terse names to retrieve `Language` instances from `String` (and not, name, short name, file extension or any other plausible `String` being part of `Language`), so it was natural to have here the logic to produce the list of supported languages… the name is hideous, but at least it kept all `String <-> Language` logic in a single place…

Reply:
This should be part of the ruleset contract I think. Alternatively, rename `getTerseName` to `getId` and canonicalize the fact that ids must be unique across languages. I really don't like the fact that we're dabbling with three different kind of "names", and that nowhere is it written that yeah, we use the terse name as an ID nearly everywhere. This is not the job of the language registry, it's just Language that's hard to use.

Reply:
This also applies to `LanguageVersion#getVersion`

Reply:
Let's not expand this PR to define the language module API... let's keep that separated.